two_factor/views/core.py

```diff
@@ -104,20 +108,31 @@ def done(self, form_list, **kwargs):
         """
         login(self.request, self.get_user())
 
-        redirect_to = self.request.POST.get(
-            self.redirect_field_name,
-            self.request.GET.get(self.redirect_field_name, '')
-        )
-
-        if not is_safe_url(url=redirect_to, allowed_hosts=[self.request.get_host()]):
-            redirect_to = resolve_url(settings.LOGIN_REDIRECT_URL)
+        redirect_to = self.get_success_url()
 
         device = getattr(self.get_user(), 'otp_device', None)
         if device:
             signals.user_verified.send(sender=__name__, request=self.request,
                                        user=self.get_user(), device=device)
         return redirect(redirect_to)
 
+    def get_success_url(self):
```

Comment:
The following lines are inspired by the main Django Project (see [Code](https://github.com/django/django/blob/4f61810751751b8c5070ce038ea57e949650e9e3/django/contrib/auth/views.py#L65)). Inheritance is, however, not an option, as inheriting the entire LoginView from the main project does not fit with the current structure.

Moreover  `is_safe_url` must be used instead of the newer `url_has_allowed_host_and_scheme` to reflect the backwards compatibility of this repo with older Django versions.

Reply:
`get_success_url`, `get_redirect_url`, and `dispatch` should each have a code comment saying that they're from Django's LoginView class and which version they're from.